### PR TITLE
Show viewpoints from BCF gallery

### PR DESCRIPTION
**Pull request by oliverguenther**
- [x] Get viewpoint from viewer
- [x] Abstract revit and xeokit services into one using `useFactory`

**TODOS**
- [x] Save viewpoint works, but does not reload work package
- [x] ~~snapshot of saved viewpoint is seemingly empty~~ Caused by incorrectly parsing the entire base64 URL, instead of only the encoded parts
- [x] Bitmaps need to be removed in frontend

### CODE REVIEW

File: frontend/src/app/modules/bim/bcf/bcf-wp-single-view/bcf-wp-single-view.component.ts [hunk 1]
```diff
@@ -2,49 +2,76 @@ import {Component, Input, OnDestroy, OnInit} from "@angular/core";
...
+import {PathHelperService} from "core-app/modules/common/path-helper/path-helper.service";
 import {HalLink} from "core-app/modules/hal/hal-link/hal-link";
+import {BcfApiService} from "core-app/modules/bim/bcf/api/bcf-api.service";
+import {BcfViewpointPaths} from "core-app/modules/bim/bcf/api/viewpoints/bcf-viewpoint.paths";
+import {CurrentProjectService} from "core-components/projects/current-project.service";
+import {ModelViewerService} from "core-app/modules/bcf/services/model-viewer.service";
+import {I18nService} from "core-app/modules/common/i18n/i18n.service";
+
+export type ViewPointOriginal = { uuid:string, snapshot_id:string, snapshot_file_name:string };
+export type ViewPoint = { snapshotId:string, snapshotFileName?:string, snapshotFullPath:string };
```
Comment (openprojectci) [inline frontend/src/app/modules/bim/bcf/bcf-wp-single-view/bcf-wp-single-view.component.ts:16]: <table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><del><span data-href="https://github.com/opf/openproject/blob/603cab1b34ef23edd1c7087100cbc18fec4969a9/frontend/src/app/modules/bim/bcf/bcf-wp-single-view/bcf-wp-single-view.component.ts#L15"></span>Please use <code>ChangeDetectionStrategy.OnPush</code> for this component</del></td>
    </tr>
  </tbody>
</table>

**Comment by oliverguenther** [comment]
**Note**: Creating a viewpoint from a newly created work package (without BCF issue) does not yet work, because the issue created always creates a work package. We depend on https://community.openproject.com/wp/32493 for this.

**Review by HDinger — COMMENTED** [review]
Looks good 👍 As said, I ignored the style issues for now. 
I found two issues, however at least one is probably the same the tests are finding: that you cannot open the split or the full screen of a WP on the WP page.

The other one happens when you open the split screen of a non-BCF Work Package in the BIM module. There are things missing, and most of the Angular buttons don't work correctly any more.
Bildschirmfoto 2020-03-10 um 13 14 47

**Comment by oliverguenther** [comment]
@HDinger I addressed all remarks. the filter spec is still failing but working locally.

**Pull request merged**